daft/recordbatch/recordbatch_io.py

```diff
@@ -617,7 +618,17 @@ def write_table() -> None:
             )
         elif file_format == FileFormat.Csv:
             output_file = fs.open_output_stream(file_path)
-            pacsv.write_csv(table, output_file)
+            if delimiter is not None and len(delimiter) == 1:
```

Comment:
Great check here! I also really like the documentation here about PyArrow's CSV writer constraining this to 1 character.